src/test/java/hudson/plugins/active_directory/ActiveDirectoryUnixAuthenticationProviderTest.java

```diff
@@ -0,0 +1,56 @@
+        assertEquals("CN=name\\,surname,OU=Jenkins-core-developer,DC=domain,DC=com", escapeCharactersInDistinguisedName(distinguisedName, false));
+
+        //Check escape of backslash character
+        distinguisedName = "CN=name\\ surname,OU=Jenkins-core-developer,DC=domain,DC=com";
+        assertEquals("CN=name\\\\\\ surname,OU=Jenkins-core-developer,DC=domain,DC=com", escapeCharactersInDistinguisedName(distinguisedName, false));
+
+        //Check escape of # character
+        distinguisedName = "CN=name#surname,OU=Jenkins-core-developer,DC=domain,DC=com";
+        assertEquals("CN=name\\#surname,OU=Jenkins-core-developer,DC=domain,DC=com", escapeCharactersInDistinguisedName(distinguisedName, false));
+
+        //Check escape of + character
+        distinguisedName = "CN=name+surname,OU=Jenkins-core-developer,DC=domain,DC=com";
+        assertEquals("CN=name\\+surname,OU=Jenkins-core-developer,DC=domain,DC=com", escapeCharactersInDistinguisedName(distinguisedName, false));
+
+        //Check escape of < character
+        distinguisedName = "CN=name<surname,OU=Jenkins-core-developer,DC=domain,DC=com";
+        assertEquals("CN=name\\<surname,OU=Jenkins-core-developer,DC=domain,DC=com", escapeCharactersInDistinguisedName(distinguisedName, false));
+
+        //Check escape of > character
+        distinguisedName = "CN=name>surname,OU=Jenkins-core-developer,DC=domain,DC=com";
+        assertEquals("CN=name\\>surname,OU=Jenkins-core-developer,DC=domain,DC=com", escapeCharactersInDistinguisedName(distinguisedName, false));
+
+        //Check escape of ; character
+        distinguisedName = "CN=name;surname,OU=Jenkins-core-developer,DC=domain,DC=com";
+        assertEquals("CN=name\\;surname,OU=Jenkins-core-developer,DC=domain,DC=com", escapeCharactersInDistinguisedName(distinguisedName, false));
+
+        //Check escape of " character
+        distinguisedName = "CN=name\"surname,OU=Jenkins-core-developer,DC=domain,DC=com";
+        assertEquals("CN=name\"surname,OU=Jenkins-core-developer,DC=domain,DC=com", escapeCharactersInDistinguisedName(distinguisedName, false));
+
+        //Check escape of = character
+        distinguisedName = "CN=name=surname,OU=Jenkins-core-developer,DC=domain,DC=com";
+        assertEquals("CN=name\\=surname,OU=Jenkins-core-developer,DC=domain,DC=com", escapeCharactersInDistinguisedName(distinguisedName, false));
+
+        //Check escape of espace character
+        distinguisedName = "CN=name surname,OU=Jenkins-core-developer,DC=domain,DC=com";
+        assertEquals("CN=name\\ surname,OU=Jenkins-core-developer,DC=domain,DC=com", escapeCharactersInDistinguisedName(distinguisedName, false));
+
+        //Check escape of forward character
+        distinguisedName = "CN=name/surname,OU=Jenkins-core-developer,DC=domain,DC=com";
+        assertEquals("CN=name\\/surname,OU=Jenkins-core-developer,DC=domain,DC=com", escapeCharactersInDistinguisedName(distinguisedName, true));
```

Comment:
if you are keeping the option for escaping slash - you should verify with the flag == `false` that the escaping doesn't happen.